testsuite/integration/expansion/src/test/java/org/wildfly/test/integration/observability/micrometer/ConflictingPrometheusContextTestCase.java

```diff
@@ -0,0 +1,126 @@
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.AssumptionViolatedException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.test.integration.observability.JaxRsActivator;
+
+@RunWith(Arquillian.class)
+@ServerSetup({MicrometerSetupTask.class})
+@DockerRequired
+@RunAsClient
+public class ConflictingPrometheusContextTestCase {
+    private static final ModelNode metricsExtension = Operations.createAddress("extension", "org.wildfly.extension.metrics");
+    private static final ModelNode metricsSubsystem = Operations.createAddress("subsystem", "metrics");
+
+    public static final ModelNode PROMETHEUS_REGISTRY_ADDRESS = Operations.createAddress(SUBSYSTEM, "micrometer", "registry", "prometheus");
+
+    private boolean metricsExtAdded = false;
+    private boolean metricsSubsystemAdded = false;
+
+    @ContainerResource
+    protected ManagementClient managementClient;
+
+    @Deployment
+    public static Archive<?> deploy() {
+        return ShrinkWrap.create(WebArchive.class, "micrometer-prometheus.war")
+                .addClasses(JaxRsActivator.class, MicrometerResource.class)
+                .addAsWebInfResource(CdiUtils.createBeansXml(), "beans.xml");
+    }
+
+    @BeforeClass
+    public static void beforeClass() {
+        if (AssumeTestGroupUtil.isBootableJar() || AssumeTestGroupUtil.isWildFlyPreview() || isGalleon()) {
+            throw new AssumptionViolatedException("Not supported in this configuration");
+        }
+    }
+
+    private static boolean isGalleon() {
+        return System.getProperty("ts.layers") != null || System.getProperty("ts.galleon") != null;
+    }
+
+    @Test
+    @InSequence(1)
+    public void setupMetrics() throws IOException {
+        if (!Operations.isSuccessfulOutcome(executeRead(managementClient, metricsExtension))) {
+            executeOp(managementClient, Operations.createAddOperation(metricsExtension));
+            metricsExtAdded = true;
+        }
+
+        if (!Operations.isSuccessfulOutcome(executeRead(managementClient, metricsSubsystem))) {
+            executeOp(managementClient, Operations.createAddOperation(metricsSubsystem));
+            metricsSubsystemAdded = true;
+        }
+
+        ServerReload.executeReloadAndWaitForCompletion(managementClient);
+    }
+
+    @Test
+    @InSequence(2)
+    public void configureConflictingContexts() throws Exception {
+        ModelNode addOperation = Operations.createAddOperation(PROMETHEUS_REGISTRY_ADDRESS);
+        addOperation.get("context").set("/metrics");
+        addOperation.get("security-enabled").set("false");
+
+        try {
+            executeOp(managementClient, addOperation);
+            Assert.fail("Adding a context that conflicts with WildFly Metrics should have failed.");
+        } catch (Exception e) {
```

Comment:
A problem here is I think if you added this test without the rest of the PR it would pass, because the op will fail with a cryptic ISE or something.

Instead calling your executeOp method you can invoke the op and work with the ModelNode returned from client.getControllerClient().execute(Operation.Factory.create(op))

ModelNode response = client.getControllerClient().execute(Operation.Factory.create(op));
AssertTrue(response.asString(), response.get(ModelDescriptionConstants.FAILURE_DESCRIPTION).asString().contains("WFLYCTL0436"));

Reply:
Done.